### PR TITLE
Pickup Points

### LINKED ISSUE #166: Routing Points/ Detours
* Feature name: Detours
* Description: Allow riders to make detours to the ride. This serves for both pickups as well as coffee runs.
* Requires:
* Status: Implementation started.
* Time Estimation: A fairly large PR. 2-3 hours of actual work. Hopefully, completed by the 7th

**Additional context**
 - For now let anyone create a detour as long as they in the ride
 - This should update the ride preview

### PR DESCRIPTION
**Pull request by doug-stormtree**
Pickup points can be added and displayed, with optimized routing between them.

Closes #166

### CODE REVIEW

**Comment by doug-stormtree** [comment]
Ride features should be working as discussed, but with limited testing so far. Will be fixing more in the morning so don't merge right away.

**Review by CobeyH — APPROVED** [review]
Certainty minor UX issues here and there but overall this is a huge improvement. Great work!

**Pull request merged**
